controllers/serviceaccount_controller_test.go

```diff
@@ -0,0 +1,252 @@
+			gardencorev1beta1.ProjectMemberServiceAccountManager,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		// make sure that there is a project-serviceaccountmanager ClusterRole
+		clusterRoleServiceAccountManager := &rbacv1.ClusterRole{ObjectMeta: metav1.ObjectMeta{Name: "gardener.cloud:system:project-serviceaccountmanager"}}
+		_, err = controllerutils.GetAndCreateOrMergePatch(ctx, e.K8sClient, clusterRoleServiceAccountManager, func() error {
+			clusterRoleServiceAccountManager.Rules = []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{""},
+					Resources: []string{"serviceaccounts"},
+					Verbs: []string{
+						"create",
+						"delete",
+						"deletecollection",
+						"get",
+						"list",
+						"patch",
+						"update",
+						"watch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"serviceaccounts/token"},
+					Verbs:     []string{"create"},
+				},
+			}
+			return nil
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		// ensure that role bindings are reconciled according to the project membership spec
+		projectRBAC, err := projectrbac.New(e.K8sClient, project)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(projectRBAC.Deploy(ctx)).To(Succeed())
+	})
+
+	JustBeforeEach(func() {
+		terminalCreationError = e.K8sClient.Create(ctx, terminal)
+	})
+
+	Context("project membership lifecycle", func() {
+		It("should remove ServiceAccount as project member", func() {
+			Expect(terminalCreationError).To(Not(HaveOccurred()))
+
+			By("Expecting terminal to be created")
+			Eventually(func() bool {
+				terminal = &dashboardv1alpha1.Terminal{}
+				err := e.K8sClient.Get(ctx, terminalKey, terminal)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			By("Adding terminal ownerReference to service account")
+			targetServiceAccount := &corev1.ServiceAccount{}
+			err := e.K8sClient.Get(ctx, targetServiceAccountKey, targetServiceAccount)
+			Expect(err).ToNot(HaveOccurred())
+			if targetServiceAccount.OwnerReferences == nil {
+				targetServiceAccount.OwnerReferences = []metav1.OwnerReference{}
+			}
+			targetServiceAccount.OwnerReferences = append(targetServiceAccount.OwnerReferences, metav1.OwnerReference{
+				APIVersion: "dashboard.gardener.cloud/v1alpha1",
+				Kind:       "Terminal",
+				Name:       terminal.Name,
+				UID:        terminal.UID,
+			})
+			Expect(e.K8sClient.Update(ctx, targetServiceAccount)).To(Succeed())
+
+			isTargetServiceAccountMember := func() bool {
+				project := &gardencorev1beta1.Project{ObjectMeta: metav1.ObjectMeta{Name: projectName}}
+				err := e.K8sClient.Get(ctx, client.ObjectKeyFromObject(project), project)
+				if err != nil {
+					return false
+				}
+
+				isMember, _ := gardenclient.IsMember(project.Spec.Members, targetServiceAccountKey)
+				return isMember
+			}
+
+			By("Ensuring that service account is member of the project")
+			Consistently(isTargetServiceAccountMember).Should(BeTrue())
```

Comment:
Why do we use Consistently here?
```suggestion
			Expect(isTargetServiceAccountMember()).To(BeTrue())
```